docs/developer-guide/getting-started.rst

```diff
@@ -0,0 +1,458 @@
+See `the GitHub documentation <https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account>`_ if you haven't done so already.
+
+.. tab-set::
+
+   .. tab-item:: Working in the main repository
+      :sync: main
+
+      Clone the repository:
+
+      .. code-block:: console
+
+          $ git clone git@github.com:cta-observatory/ctapipe.git
+          $ cd ctapipe
+
+
+   .. tab-item:: Working a fork
+      :sync: fork
+
+      In order to checkout the software so that you can push changes to GitHub without
+      having write access to the main repository at ``cta-observatory/ctapipe``, you
+      `need to fork <https://help.github.com/articles/fork-a-repo/>`_ it.
+
+      After that, clone your fork of the repository and add the main reposiory as a second
+      remote called ``upstream``, so that you can keep your fork synchronized with the main repository.
+
+      .. code-block:: console
+
+          $ git clone https://github.com/[YOUR-GITHUB-USERNAME]/ctapipe.git
+          $ cd ctapipe
+          $ git remote add upstream https://github.com/cta-observatory/ctapipe.git
+
+
+
+Setting up the development environment
+--------------------------------------
+
+We provide a conda environment with all packages needed for development of ctapipe and a couple of additonal helpful pacakages (like ipython, jupyter and vitables):
+
+.. code-block:: console
+
+    $ mamba env create -f environment.yml
+
+
+Next, switch to this new virtual environment:
+
+.. code-block:: console
+
+    $ mamba activate cta-dev
+
+You will need to run that last command any time you open a new
+terminal to activate the conda environment.
+
+
+Installing ctapipe in development mode
+--------------------------------------
+
+Now setup this cloned version for development.
+The following command will use the editable installation feature of python packages.
+From then on, all the ctapipe executables and the library itself will be
+usable from anywhere, given you have activated the ``cta-dev`` conda environment.
+
+.. code-block:: console
+
+    $ pip install -e .
+
+Using the editable installation means you won't have to rerun the installation for
+simple code changes to take effect.
+However, for things like adding new submodules or new entry points, rerunning the above
+step might still be needed.
+
+ctapipe supports adding new ``EventSource`` and ``Reconstructor`` implementations
+through plugins. In order for the respective tests to pass you have to install the
+test plugin via
+
+.. code-block:: console
+
+    $ pip install -e ./test_plugin
+
+
+We are using the ``black`` and ``isort`` auto-formatters for automatic 
+adherence to the code style (see our :doc:`/developer-guide/style-guide`).
+To enforce running these tools whenever you make a commit, setup the
+`pre-commit hook <https://pre-commit.com/>`_:
+
+.. code-block:: console
+
+    $ pre-commit install
+
```

Comment:
Suggested addition about how the pre-commit hook will block your commit. Something like:

> The pre-commit hook will then execute the tools with the same settings as when the a pull request is checked on github, and if any problems are reported the commit will be rejected. You then have to fix the reported issues before tying to commit again.
>     Note that a common problem is forgetting to run the black code formatter, and that whenever this was the only problem found, simply adding the changes resulting from the pre-commit hook to the commit will result in your changes being accepted.
>

Reply:
Done!